packages/react-portal-compat/src/PortalCompatProvider.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { useThemeClassName } from '@fluentui/react-components';
+
+import { PortalCompatContext } from './PortalCompatContext';
+import { RegisterPortalFn } from './types';
+
+export const PortalCompatProvider: React.FC = props => {
+  const { children } = props;
+
+  const themeClassName = useThemeClassName();
+  const cssVariablesClassName = React.useMemo<string | undefined>(
+    // "themeClassName" may contain multiple classes while we want to add only a class that hosts CSS variables
+    // Keep in sync with "packages/react-provider/src/components/FluentProvider/useThemeStyleTag.ts"
```

Comment:
I assume this is not being exported from react-components right, thus the need of duplication ?

Reply:
Before there was dependency only on `shared-contexts`, I will check if we can use a const from `react-components`

Reply:
```
const CLASS_NAME_REGEX = new RegExp(`(${fluentProviderClassNames.root}\\d+)`);
```

This regexp now uses `fluentProviderClassNames` so it should be better than before. Anyway I left the comment as logic might change in `useFluentProviderThemeStyleTag.ts`.